### PR TITLE
Polishing

### PR DESCRIPTION
**Pull request by jbduncan**
## Overview

This PR is a smaller separated part of #1049, as requested by @sbrannen. It does the following:
- Apply `private` and `final` to an applicable class field.
- Remove usages of `Collectors.toList()` and use `Stream` directly for brevity.
- Use AssertJ's assertions over JUnit 5's more for consistency and clarity.

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 1]
```diff
@@ -69,14 +68,14 @@ void packageNameFilterInclude_matchingPackagesAreIncluded() {
...
 		applier.applyAllFilters(request, engineDescriptor);
 
-		List<UniqueId> includedDescriptors = engineDescriptor.getDescendants().stream().map(
-			TestDescriptor::getUniqueId).collect(Collectors.toList());
-		Assertions.assertEquals(1, includedDescriptors.size());
-		Assertions.assertTrue(includedDescriptors.contains(UniqueId.root("class", "matching")));
+		Stream<UniqueId> includedDescriptors =
+				engineDescriptor.getDescendants().stream().map(TestDescriptor::getUniqueId);
+		assertThat(includedDescriptors)
+				.containsExactly(UniqueId.root("class", "matching"));
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]:
```java
assertThat(engineDescriptor.getDescendants())
    .extracting(TestDescriptor::getUniqueId)
    .containsExactly(UniqueId.root("class", "matching"))
```

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 2]
```diff
@@ -91,13 +90,13 @@ void packageNameFilterExclude_matchingPackagesAreExcluded() {
...
-		// @formatter:on
 
 		applier.applyAllFilters(request, engineDescriptor);
 
-		List<UniqueId> includedDescriptors = engineDescriptor.getDescendants().stream().map(
-			TestDescriptor::getUniqueId).collect(Collectors.toList());
-		Assertions.assertEquals(0, includedDescriptors.size());
+		Stream<UniqueId> includedDescriptors =
+				engineDescriptor.getDescendants().stream().map(TestDescriptor::getUniqueId);
+		assertThat(includedDescriptors).isEmpty();
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Ditto.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 3]
```diff
@@ -112,14 +111,14 @@ void packageNameFilterExclude_nonMatchingPackagesAreIncluded() {
...
 		applier.applyAllFilters(request, engineDescriptor);
 
-		List<UniqueId> includedDescriptors = engineDescriptor.getDescendants().stream().map(
-			TestDescriptor::getUniqueId).collect(Collectors.toList());
-		Assertions.assertEquals(1, includedDescriptors.size());
-		Assertions.assertTrue(includedDescriptors.contains(UniqueId.root("class", "matching")));
+		Stream<UniqueId> includedDescriptors =
+				engineDescriptor.getDescendants().stream().map(TestDescriptor::getUniqueId);
+		assertThat(includedDescriptors)
+				.containsExactly(UniqueId.root("class", "matching"));
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Ditto.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 4]
```diff
@@ -130,19 +129,19 @@ void nonMatchingClassesAreExcluded() {
...
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Ditto.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 5]
```diff
@@ -152,20 +151,21 @@ void nestedTestClassesAreAlwaysIncludedWhenTheirParentIs() {
...
-			TestDescriptor::getUniqueId).collect(Collectors.toList());
-		Assertions.assertEquals(2, includedDescriptors.size());
-		Assertions.assertTrue(includedDescriptors.contains(UniqueId.root("class", "matching")));
-		Assertions.assertTrue(includedDescriptors.contains(UniqueId.root("nested-class", "nested")));
+		Stream<UniqueId> includedDescriptors =
+				engineDescriptor.getDescendants().stream().map(TestDescriptor::getUniqueId);
+		assertThat(includedDescriptors).containsExactlyInAnyOrder(
+				UniqueId.root("class", "matching"),
+				UniqueId.root("nested-class", "nested")
+		);
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Ditto.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 6]
```diff
@@ -48,13 +47,13 @@ void packageNameFilterInclude_nonMatchingPackagesAreExcluded() {
...
-		// @formatter:on
 
 		applier.applyAllFilters(request, engineDescriptor);
 
-		List<UniqueId> includedDescriptors = engineDescriptor.getDescendants().stream().map(
-			TestDescriptor::getUniqueId).collect(Collectors.toList());
-		Assertions.assertEquals(0, includedDescriptors.size());
+		Stream<UniqueId> includedDescriptors =
+				engineDescriptor.getDescendants().stream().map(TestDescriptor::getUniqueId);
+		assertThat(includedDescriptors).isEmpty();
```
[the remark this answers is not in the document]
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Yes, I think you're right... Makes me wonder why we were mapping from descendants to `UniqueId`s here.

@stefanbirkner, can you remember why it was programmed this way?

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 1 above)
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Good catch! Thanks @marcphilipp. :)

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: I've pushed a new commit (https://github.com/junit-team/junit5/pull/1051/commits/dfb29805fae1fb0a0f5d8609b233bf7e7e36a4dc) with @marcphilipp's suggested change here on the assumption that mapping from descendants to `UniqueId`s is _not_ strictly needed to test for emptyness.

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 2 above)
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Done. :+1:

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 1 above)
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Done. :+1:

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 3 above)
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Done :+1:

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 4 above)
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Done. :+1:

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java (hunk 5 above)
Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Done. :+1:

**Comment by jbduncan** [comment]
@marcphilipp Many thanks for your feedback! I believe I've addressed all of your comments now. Please let me know if there's anything else to do. :)

**Comment by jbduncan** [comment]
Formatting fixed. :P

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java [hunk 7]
```diff
@@ -48,13 +45,11 @@ void packageNameFilterInclude_nonMatchingPackagesAreExcluded() {
...
 				.build();
-		// @formatter:on
 
 		applier.applyAllFilters(request, engineDescriptor);
 
-		List<UniqueId> includedDescriptors = engineDescriptor.getDescendants().stream().map(
-			TestDescriptor::getUniqueId).collect(Collectors.toList());
-		Assertions.assertEquals(0, includedDescriptors.size());
+		assertThat(engineDescriptor.getDescendants()).isEmpty();
+		// @formatter:on
```
Comment (marcphilipp) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: I think you can move these back up now, at least the ones for "empty".

Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Hi @marcphilipp, can you clarify for me what you mean by "move these back up"? Would you like me to remove the blank line between `applier.applyAllFilters(request, engineDescriptor);` and `assertThat(engineDescriptor.getDescendants()).isEmpty();` (and so the same for all the other "empty" assertions)? Or have I misunderstood your request?

Comment (marcphilipp) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: I was referring to the line I commented on `// @formatter:on`.

Comment (jbduncan) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/DiscoveryFilterApplierTests.java]: Oh, I see now. Silly me! Thanks @marcphilipp. :)

**Comment by jbduncan** [comment]
@marcphilipp, I've applied further feedback which you kindly clarified for me at https://github.com/junit-team/junit5/pull/1051#discussion_r138336511. :)

**Comment by marcphilipp** [comment]
Thanks! 🙂

**Comment by jbduncan** [comment]
You're very welcome @marcphilipp! :+1:

**Pull request merged**